src/test/kotlin/com/exactpro/th2/check1/rule/sequence/TestSequenceCheckTask.kt

```diff
@@ -111,10 +112,32 @@ class TestSequenceCheckTask : AbstractCheckTaskTest() {
 
         sequenceCheckRuleTask(parentEventID, messageStream, checkOrder).begin()
 
-        val batchRequest = awaitEventBatchRequest(1000L)
-        val eventsList: List<Event> = batchRequest.eventsList
+        val batchRequest = awaitEventBatchRequest(1000L, 5)
+        val eventsList: List<Event> = batchRequest.flatMap(EventBatch::getEventsList)
 
         assertAll({
+            with(batchRequest[0]) {
+                assertEquals(1, eventsCount)
+                assertEquals("checkSequenceRule", getEvents(0).type)
+            }
+            with(batchRequest[1]) {
+                assertEquals(1, eventsCount)
+                assertEquals("preFiltering", getEvents(0).type)
+            }
+            with(batchRequest[2]) {
+                assertEquals(3, eventsCount)
+                assertTrue (getEventsList().all { "Verification" == it.type })
+            }
+            with(batchRequest[3]) {
+                assertEquals(2, eventsCount)
+                assertNotNull (getEventsList().firstOrNull { "checkSequence" == it.type })
+                assertNotNull (getEventsList().firstOrNull { "checkMessages" == it.type })
+            }
+            with(batchRequest[4]) {
```

Comment:
I think we need to check the `batchRequest` size before getting the events by index

Reply:
the awaitEventBatchRequest(timeout, times) method already check it